packages/cli/src/commands/screenshot-diff/screenshot-diff.command.ts

```diff
@@ -18,199 +22,212 @@ import {
-      logComparisonResultMessage(
-        logger,
-        `${Math.round(
-          result.comparisonResult.mismatchFraction * 100
-        )}% pixel mismatch for screenshot ${basename(
-          result.headScreenshotFile
-        )} (threshold is ${Math.round(diffThreshold * 100)}%)`,
-        result.outcome
-      );
-    });
-
-    // Check if individual screenshot mismatch is higher than the threshold.
-    const mismatchingScreenshots = comparisonResults.filter(
-      (result) => result.outcome == "fail"
-    );
-
-    if (mismatchingScreenshots.length) {
-      logger.info(
-        `Screenshots ${mismatchingScreenshots
-          .map((result) => basename(result.headScreenshotFile))
-          .sort()} do not match!`
-      );
-      if (exitOnMismatch) {
-        process.exit(1);
-      }
-      throw new DiffError(
-        `Screenshots ${mismatchingScreenshots.map((result) =>
-          basename(result.headScreenshotFile)
-        )} do not match!`,
-        {
-          baseReplayId,
-          headReplayId,
-          threshold: diffThreshold,
+          if (!baseReplayScreenshots.includes(screenshotFileName)) {
+            return [
+              {
+                identifier,
+                outcome: "missing-base",
+                headScreenshotFile: `screenshots/${screenshotFileName}`,
+              },
+            ];
+          }
+
+          const baseScreenshotFile = join(
+            baseScreenshotsDir,
+            screenshotFileName
+          );
+          const headScreenshotFile = join(
+            headScreenshotsDir,
+            screenshotFileName
+          );
+          const baseScreenshot = await readPng(baseScreenshotFile);
+          const headScreenshot = await readPng(headScreenshotFile);
+
+          try {
+            const comparisonResult = compareImages({
+              base: baseScreenshot,
+              head: headScreenshot,
+              pixelThreshold: diffPixelThreshold,
+            });
+
+            await writeScreenshotDiff({
+              baseReplayId,
+              headReplayId,
+              screenshotFileName,
+              diff: comparisonResult.diff,
+            });
+
+            return [
+              {
+                identifier,
+                outcome:
+                  comparisonResult.mismatchFraction > diffThreshold
+                    ? "diff"
+                    : "no-diff",
+                headScreenshotFile: `screenshots/${screenshotFileName}`,
+                baseScreenshotFile: `screenshots/${screenshotFileName}`,
+                width: baseScreenshot.width,
+                height: baseScreenshot.height,
+                mismatchPixels: comparisonResult.mismatchPixels,
+                mismatchFraction: comparisonResult.mismatchFraction,
+              },
+            ];
+          } catch (error) {
+            if (
+              error instanceof Error &&
+              error.message.startsWith("Cannot handle different size")
```

Comment:
[nit / not important] May be cleaner to compare the sizes ourselves to check this, or to have compareImages return a type union rather than throw an error, or use a shared constant or custom error type. That way this wouldn't break if we change the text of their error message slightly / non-obvious in compareImages that changing that text may break things.

Reply:
Done.